frontend/src/components/input.tsx

```diff
@@ -52,26 +58,59 @@ export const Input = ({ sendMessage, waiting, suggestions }: InputProps) => {
     [sendMessage, userInput, waiting],
   );
 
+  const uploadFile = async (file: File) => {
+    setIsUploading(true);
+    setIsUploaded(false);
+
+    try {
+      const formData = new FormData();
+      formData.append('file', file);
+      const response = await fetch(`${process.env.BACKEND_URL}/uploadfile`, {
+        method: 'POST',
+        body: formData,
+        credentials: 'include',
+      });
+      if (!response.ok)
+        throw new Error(`Upload failed with status ${response.status}`);
+
+      const { filename, id } = await response.json();
+      console.log(`File uploaded successfully: ${filename} with id ${id}`);
+      // TODO: Set fileId in the state
+      // setFileId(id);
+      setUploadedFile(file);
+      setIsUploading(false);
+      setIsUploaded(true);
```

Comment:
Do we need the `isUploaded` state? We could maybe just use the fact the `uploadedFile` is set?

Reply:
Yes, you are right. Initially I used it for quick work, but then forgot to use the uploadedFile and remove it.